solr/core/src/java/org/apache/solr/update/processor/NumFieldsMonitor.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.update.processor;
+
+import org.apache.solr.core.AbstractSolrEventListener;
+import org.apache.solr.core.SolrCore;
+import org.apache.solr.search.SolrIndexSearcher;
+import org.apache.solr.util.RefCounted;
+
+public class NumFieldsMonitor extends AbstractSolrEventListener {
+
+  private int numFields;
+
+  public NumFieldsMonitor(SolrCore core) {
+    super(core);
+    this.numFields = -1;
+  }
+
+  @Override
+  public void postCommit() {
+    RefCounted<SolrIndexSearcher> indexSearcher = null;
+    try {
+      indexSearcher = getCore().getSearcher();
```

Comment:
see SolrCore.withSearcher to make this code less involved.  I added that long ago and getSearcher's javadocs recommend using it.

Reply:
Is there a reason that `withSearcher` throws an IOException?  Would it be worth having one of those methods that consumes a vanilla Function instead of a `IOFunction`?

Reply:
Maybe; Java exception handling is such a PITA.... I didn't want to defeat the lambda convenience if the lambda threw an IOException which is rather common.